.generator/cli.py

```diff
@@ -501,6 +500,47 @@ def _generate_repo_metadata_file(
     _write_json_file(output_repo_metadata, metadata_content)
 
 
+def _copy_readme_to_docs(output: str, library_id: str):
+    """Copies the README.rst file for a generated library to docs/README.rst.
+
+    This function is robust against various symlink configurations that could
+    cause `shutil.copy` to fail with a `SameFileError`. It reads the content
+    from the source and writes it to the destination, ensuring the final
+    destination is a real file.
+
+    Args:
+        output(str): Path to the directory in the container where code
+            should be generated.
+        library_id(str): The library id.
+    """
+    path_to_library = f"packages/{library_id}"
+    source_path = f"{output}/{path_to_library}/README.rst"
+    docs_path = f"{output}/{path_to_library}/docs"
+    destination_path = f"{docs_path}/README.rst"
+
+    # If the source file doesn't exist (not even as a broken symlink),
+    # there's nothing to copy.
+    if not os.path.lexists(source_path):
+        return
+
+    # Read the content from the source, which will resolve any symlinks.
+    with open(source_path, "r") as f:
+        content = f.read()
```

Comment:
nit: use the `_read_text_file` function

https://github.com/googleapis/google-cloud-python/blob/79a526115b35fec0df48ca9512673eb5b96c641e/.generator/cli.py#L61-L72